tests/jwt_test.py

```diff
@@ -4,50 +4,48 @@
 import tempfile
 import json
 import time
-import base64
-from http.server import HTTPServer, BaseHTTPRequestHandler
-from http import HTTPStatus
-import ssl
-import threading
-from contextlib import AbstractContextManager
 import infra.network
 import infra.path
 import infra.proc
 import infra.net
 import infra.e2e_args
 import suite.test_requirements as reqs
+import infra.jwt
 
 from loguru import logger as LOG
 
 this_dir = os.path.dirname(__file__)
 
 
-def create_jwks(kid, cert_pem, test_invalid_is_key=False):
-    der_b64 = base64.b64encode(
-        infra.crypto.cert_pem_to_der(cert_pem)
-        if not test_invalid_is_key
-        else infra.crypto.pub_key_pem_to_der(cert_pem)
-    ).decode("ascii")
-    return {"keys": [{"kty": "RSA", "kid": kid, "x5c": [der_b64]}]}
+@reqs.description("Refresh JWT issuer")
+def test_refresh_jwt_issuer(network, args):
+    assert network.jwt_issuer.server, "JWT server is not started"
+    jwt_server = network.jwt_issuer.server
```

Comment:
I'm sure there's a reason, but why is restarting the server necessary? Shouldn't the job of `refresh_keys` also be to refresh the jwks?

Reply:
I only did this this to avoid race conditions as the OpenID server listens on a separate thread, and refreshing the jwks while the server is serving those may (very unlikely but still) cause CI instabilities.

Reply:
Global Interpreter Lock :)